### PR TITLE
Mer dashboard

### PR DESCRIPTION
**Pull request by Ashkanthegreat**
- Make Merchant dashboard
- Have link to merchant's pending orders and order details on merchant dashboard
- As a Merchant's employee, I'll be able to my Merchant's items
- Make it so a Admin can see the a merchant's dashboard

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -30,4 +30,18 @@ def packaged_check
...
+    Merchant.joins(:orders)
+          .where("merchants.id = ?", merchant_id)
+          .where("orders.id = ?", id)
+          .sum("item_orders.quantity")
+  end
+
+  def total_merchant_value(merchant_id)
+    Merchant.joins(:orders)
+          .where("merchants.id = ?", merchant_id)
+          .where("orders.id = ?", id)
```
Comment (jpc20) [inline app/models/order.rb:44]: If we had more time we could consolidate the first 3 lines to a helper method

File: app/models/merchant.rb [hunk 2]
```diff
@@ -1,6 +1,8 @@
 class Merchant <ApplicationRecord
   has_many :items, dependent: :destroy
   has_many :item_orders, through: :items
+  has_many :orders, through: :item_orders
+  has_many :users, -> { where(role: 1) }
```
Comment (Lithnotep) [inline app/models/merchant.rb:5]: Ash you knew what you were doing! Shouldn't have doubted you. has many relation was the way to go

File: app/models/user.rb [hunk 3]
```diff
@@ -6,4 +6,5 @@ class User < ApplicationRecord
   has_many :orders
   has_secure_password
   enum role: %w(default merchant admin)
+  belongs_to :merchant, optional: true
```
Comment (jpc20) [inline app/models/user.rb:9]: Users now can belong to a merchant so they can log in and act as that merchant

File: spec/features/users/user_admin_spec.rb [hunk 4]
```diff
@@ -1,6 +1,22 @@
 require 'rails_helper'
 
 RSpec.describe "As a Admin" do
+
```
Comment (janegreene) [inline spec/features/users/user_admin_spec.rb:4]: Thanks for streamlining this test.

**Pull request merged**